### ISSUE TITLE
PHP Fatal error:  Uncaught Phpro\SoapClient\Exception\InvalidArgumentException: The classmap already exists!

### ISSUE THREAD

**Issue by 4n70w4** [phpro/soap-client#235]
### Bug Report

Environment see in https://github.com/phpro/soap-client/issues/233

#### Current behavior

Do not work(

#### How to reproduce

`./vendor/bin/soap-client wizard -vvv`

 ```
 Where would you like to store your config file? [config/soap-client.php]:
 >


 Wsdl location (URL or path to file):
 > https://service.synerdocs.ru/ExchangeService.svc?wsdl

 Generic name used to name this client (Results in <name>Client <name>Classmap etc.):
 > Synerdocs

 Directory where the client should be generated in:
 > app/Integrations/Synerdocs

 Namespace for your client:
 > App/Integrations/Synerdocs

 This tool can set some basic code generation rules. This requires some knowledge of the SOAP service.
Take a look at the message section in the WSDL. Are you able to match request and response elements based on keywords?
These keywords are used in a case insensitive regex match with '/' delimiters, escape accordingly! (yes/no) [no]:
 >
```

` [OK] Config has been written to config/soap-client.php`

`Generated class ........`

```
 [OK] All SOAP types generated

 [OK] Generated classmap at app/Integrations/Synerdocs/SynerdocsClassmap.php

 [OK] Generated client at app/Integrations/Synerdocs/SynerdocsClient.php

 [OK] Generated client factory at app/Integrations/Synerdocs/SynerdocsClientFactory.php
```

Create api.php

```
<?php

require_once('vendor/autoload.php');

$client = \App\Integrations\Synerdocs\SynerdocsClientFactory::factory('https://service.synerdocs.ru/ExchangeService.svc?wsdl');

$response = $client->authenticate((new \App\Integrations\Synerdocs\Type\Authenticate())->withLogin('something')->withPassword('something')->withApplicationId('something') );
echo $response->getGreeting();
```

`php api.php`

```
PHP Fatal error:  Uncaught Phpro\SoapClient\Exception\InvalidArgumentException: The classmap already exists! in /app/vendor/phpro/soap-client/src/Phpro/SoapClien
t/Soap/ClassMap/ClassMapCollection.php:39
Stack trace:
#0 /app/vendor/phpro/soap-client/src/Phpro/SoapClient/Soap/ClassMap/ClassMapCollection.php(27): Phpro\SoapClient\Soap\ClassMap\ClassMapCollection->add(Object(Php
ro\SoapClient\Soap\ClassMap\ClassMap))
#1 /app/app/Integrations/Synerdocs/SynerdocsClassmap.php(768): Phpro\SoapClient\Soap\ClassMap\ClassMapCollection->__construct(Array)
#2 /app/app/Integrations/Synerdocs/SynerdocsClientFactory.php(18): App\Integrations\Synerdocs\SynerdocsClassmap::getCollection()
#3 /app/api.php(5): App\Integrations\Synerdocs\SynerdocsClientFactory::factory('https://service...')
#4 {main}
  thrown in /app/vendor/phpro/soap-client/src/Phpro/SoapClient/Soap/ClassMap/ClassMapCollection.php on line 39
```

#### Expected behavior

Successful work with api.

**Comment by veewee** [comment]
It looks like there are duplicate types in your classmap.
Working with duplicate types is a known issue in ext-soap.
Read more about it here: https://github.com/phpro/soap-client/blob/master/docs/known-issues/ext-soap.md#duplicate-typenames

**Comment by 4n70w4** [comment]
@veewee lol, you have not solve request, but closed it? :))
In another project, such an error does not occur, they were able to solve it.
https://github.com/WsdlToPhp/PackageGenerator

And ext-soap is also used there.

**Comment by veewee** [comment]
Hello @4n70w4 

The issue was closed since I provided you with the details on how to resolve your problem. 

Since the package you referenced to is using ext-soap with classmaps, it suffers from exactly the same issue. However, that package does not throw an exception, which might give you a false feeling that the code is mapped correctly.  (see the link I provided above for test case and links to php-src)

I agree that it would be better if the code warns you of this during code generation instead of a hard exception. I'll see what I can do to improve developer experience in the short term.

Handling duplicate type names is something we are planning to improve in the long term. 
We don't know yet if we well be providing workarounds for this, or provide additional userland soap drivers which do support this feature.
